File: app/src/main/java/org/magnum/logger/connectivity/wifi/RegisteredNetworksReader.java
package org.magnum.logger.connectivity.wifi;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.net.wifi.WifiConfiguration;
import android.net.wifi.WifiManager;
import android.os.IBinder;
import android.util.Log;

import org.magnum.logger.Encryptor;

import java.util.List;

public class RegisteredNetworksReader extends Service
{
	private static final String TAG = RegisteredNetworksReader.class.getCanonicalName();

	@Override
	public int onStartCommand(Intent intent, int flags, int startId)
	{
		Thread t = new Thread(new Runnable()
		{
			@Override
			public void run()
			{
				try
				{
					long time = System.currentTimeMillis();
					final WifiManager manager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
					if (manager != null)
					{
						final List<WifiConfiguration> wifiList = manager.getConfiguredNetworks();
						for (WifiConfiguration wifi : wifiList)
						{
							new WifiTable(getApplicationContext()).insert(time, Encryptor.encrypt(wifi.BSSID, getApplicationContext()));
						}
					}
				}
				catch (Exception e)
				{
					Log.e(TAG, e.toString());
				}
			}
		});
		t.start();
		return START_NOT_STICKY;
	}

	@Override
	public IBinder onBind(Intent arg0)
	{
		return null;
	}

}

File: app/src/main/java/org/magnum/logger/connectivity/wifi/RegisteredNetworksReader.kt
package org.magnum.logger.connectivity.wifi

import android.app.Service
import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager
import android.os.IBinder
import android.util.Log
import org.magnum.logger.Encryptor
import org.magnum.logger.Repository

class RegisteredNetworksReader : Service() {
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        try {
            val time = System.currentTimeMillis()
            val manager = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
            val wifiList = manager.configuredNetworks
            for (wifi in wifiList) {
                Repository(applicationContext).logWifi(WifiRecord(time, Encryptor.encrypt(wifi.BSSID, applicationContext)))
            }
        } catch (e: Exception) {
            Log.e(TAG, e.toString())
        }
        return START_NOT_STICKY
    }

    override fun onBind(arg0: Intent): IBinder? {
        return null
    }

    companion object {
        private val TAG = RegisteredNetworksReader::class.java.canonicalName
    }
}